main/src/ca/uwaterloo/flix/language/phase/CodeGen.scala

```diff
@@ -630,11 +630,65 @@ object CodeGen extends Phase[ExecutableAst.Root, ExecutableAst.Root]{
       // Invoking the constructor
       visitor.visitMethodInsn(INVOKESPECIAL, decorate(clazzName), "<init>", s"(${desc})V", false)
 
-    case Expression.Ref(exp, tpe, loc) => ??? // TODO
+    case Expression.Ref(exp, tpe, loc) =>
+      // Adding source line number for debugging
+      addSourceLine(visitor, loc)
+      // Class appropriate reference class
+      val clazz = getReferenceClazz(tpe)
+      // Get internal name of the class
+      val clazzInternalName = asm.Type.getInternalName(clazz)
+      // Get constructor of the class
+      val constructorDescriptor = asm.Type.getConstructorDescriptor(clazz.getConstructors.last)
```

Comment:
Why is it last? I would expect it to be first. Even if the list only contains one constructor.